src/lib/proxy/proxy-repository.ts

```diff
@@ -85,11 +88,19 @@ export class ProxyRepository
 
     stop(): void {
         this.stores.eventStore.off(ANY_EVENT, this.onAnyEvent);
-        clearTimeout(this.timer);
+        this.running = false;
     }
 
     private async dataPolling() {
         this.timer = setTimeout(async () => {
+            if (!this.running && this.timer) {
+                clearTimeout(this.timer);
+                this.timer = null;
+                this.logger.debug(
+                    'Shutting down data polling for proxy repository',
+                );
```

Comment:
**Alternative idea**: you can have an optional callback function as a parameter. 
This callback function will never be injected, except for the test code.
In the test setup you can inject `jest.fn()` and expect it to be called once

Reply:
I really like the approach there but it doesn't actually catch the issue on main and it seems to break other tests :( I've reset that change out of the branch but I have a copy on my machine if we want to build on it

Reply:
Yeah, let's go with the improvement